bundles/core/org.eclipse.smarthome.core.persistence/src/main/java/org/eclipse/smarthome/core/persistence/internal/PersistenceManagerImpl.java

```diff
@@ -80,31 +81,35 @@ public PersistenceManagerImpl() {
 
     protected void activate() {
         scheduler = ExpressionThreadPoolManager.getExpressionScheduledPool("persist");
+        allItemsChanged(null);
+        started = true;
+        itemRegistry.addRegistryChangeListener(this);
     }
 
     protected void deactivate() {
-        scheduler.shutdown();
+        itemRegistry.removeRegistryChangeListener(this);
+        started = false;
+        removeTimers();
         scheduler = null;
     }
 
     @Reference
     protected void setItemRegistry(ItemRegistry itemRegistry) {
         this.itemRegistry = itemRegistry;
-        itemRegistry.addRegistryChangeListener(this);
-        allItemsChanged(null);
     }
 
     protected void unsetItemRegistry(ItemRegistry itemRegistry) {
-        itemRegistry.removeRegistryChangeListener(this);
         this.itemRegistry = null;
     }
 
     @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
     protected void addPersistenceService(PersistenceService persistenceService) {
         logger.debug("Initializing {} persistence service.", persistenceService.getId());
         persistenceServices.put(persistenceService.getId(), persistenceService);
-        stopEventHandling(persistenceService.getId());
-        startEventHandling(persistenceService.getId());
+        if (started) {
```

Comment:
As we are using a dynamic reference the references can be bound before and after activation.
If the persistence service are bound in front of activation, at which place the event handling is started?
Shouldn't the activate method contain a `startEventHandling` for all persistence service IDs and the deactivate method a `stopEventHandling` for all persistence service IDs.

Reply:
I had checked that - the `addConfig` method takes care of calling `startEventHandling` for every config that exists, so there is no need to do this in `activate`.
The `deactivate` method calls ´removeTimers`, which is identical to removing event handling for all services at once. What it might be missing in general is to call `removeStateChangeListener` on all items where it had added itself - but this seems to be yet another bug in the code, which I have just fixed with another commit.

Reply:
I missed a method "start event handling for all existing configurations" on the activate method. But I assume we can rely on the fact that a component is not reused and `addConfig` can only be called on an already activated method. So, all fine...